nixos/modules/services/web-apps/alps.nix

```diff
@@ -0,0 +1,99 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.alps;
+in {
+  options.services.alps = {
+    enable = mkEnableOption "alps";
+
+    port = mkOption {
+      type = types.port;
+      default = 1323;
+      description = ''
+        TCP port the process should listen to.
+      '';
+    };
+
+    bindIP = mkOption {
+      default = "[::]";
+      type = types.str;
+      description = ''
+        IP the service should listen to.
+      '';
+    };
+
+    theme = mkOption {
+      type = types.enum [ "alps" "sourcehut" ];
+      default = "sourcehut";
+      description = ''
+        The frontend's theme.
+      '';
+    };
+
+    imaps = {
+      port = mkOption {
+        type = types.port;
+        default = 993;
+        description = ''
+          The imaps port of the server
+        '';
+      };
+
+      host = mkOption {
+        type = types.str;
+        default = "[::1]";
+        example = "mail.example.org";
+        description = ''
+          The imaps server's address.
+        '';
+      };
+    };
+
+    smtps = {
+      port = mkOption {
+        type = types.port;
+        default = 445;
+        description = ''
+          The smtps port of the server
+        '';
+      };
+
+      host = mkOption {
+        type = types.str;
+        default = cfg.imaps.host;
+        defaultText = "services.alps.imaps.host";
+        example = "mail.example.org";
+        description = ''
+          The smtps server's address.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.alps = {
+      description = "alps is a simple and extensible webmail.";
+      documentation = [ "https://git.sr.ht/~migadu/alps" ];
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target" ];
+
+      path = with pkgs; [ coreutils ];
```

Comment:
```suggestion
```

Reply:
Sorry for lacking the understanding. Why is coreutils not needed here?

Reply:
coreutils is in the default PATH